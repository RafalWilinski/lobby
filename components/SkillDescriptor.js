import React from "react";
import { Select, Form, Rate } from "antd";
import skills from "../consts/skills";

const FormItem = Form.Item;
const Option = Select.Option;

class SkillDescriptor extends React.Component {
  handleRateChange = value => {
    this.setState({ value });
  };

  render() {
    const { key, index } = this.props;
    const { getFieldDecorator, getFieldValue } = this.props.form;

    const formItemLayout = {
      labelCol: {
        xs: { span: 24 },
        sm: { span: 4 }
      },
      wrapperCol: {
        xs: { span: 24 },
        sm: { span: 14 }
      }
    };

    const formItemLayoutWithOutLabel = {
      wrapperCol: {
        xs: { span: 24, offset: 0 },
        sm: { span: 14, offset: 4 }
      }
    };

    return (
      <div>
        <FormItem
          required={true}
          {...(index === 0 ? formItemLayout : formItemLayoutWithOutLabel)}
          label={index === 0 ? "Umiejętnosc" : ""}
          style={{ marginBottom: "10px" }}
        >
          {getFieldDecorator(`skill-name-${key}`, {
            validateTrigger: ["onChange", "onBlur"],
            rules: [
              {
                required: true,
                whitespace: true,
                message: "Proszę podaj umiejętnosc"
              }
            ]
          })(
            <Select
              showSearch
              style={{ width: 200 }}
              placeholder="Wybierz umiejętnosc"
              optionFilterProp="children"
              filterOption={(input, option) =>
                option.props.children
                  .toLowerCase()
                  .indexOf(input.toLowerCase()) >= 0
              }
            >
              {skills.map(skill => (
                <Option key={skill} value={skill}>
                  {skill}
                </Option>
              ))}
            </Select>
          )}
        </FormItem>
        <FormItem
          required={true}
          {...(index === 0 ? formItemLayout : formItemLayoutWithOutLabel)}
          label={index === 0 ? "Stopien Zaawansowania" : ""}
        >
          {getFieldDecorator(`skill-value-${key}`, {
            validateTrigger: ["onChange", "onBlur"],
            rules: [
              {
                required: true,
                whitespace: true,
                message: "Podaj stopień zaawansowania"
              }
            ]
          })(
            <span>
              <Rate />
            </span>
          )}
        </FormItem>
      </div>
    );
  }
}

export default SkillDescriptor;
