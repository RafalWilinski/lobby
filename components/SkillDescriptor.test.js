import React from "react";
import renderer from "react-test-renderer";
import SkillDescriptor from "./SkillDescriptor";

describe("SkillDescriptor", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<SkillDescriptor form={{ getFieldDecorator: () => () => {} }} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
