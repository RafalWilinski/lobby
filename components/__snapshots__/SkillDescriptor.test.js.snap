// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillDescriptor renders correctly 1`] = `
<div>
  <div
    className="ant-row ant-form-item"
    style={
      Object {
        "marginBottom": "10px",
      }
    }
  >
    <div
      className="ant-form-item-control-wrapper ant-col-xs-24 ant-col-xs-offset-0 ant-col-sm-14 ant-col-sm-offset-4"
    >
      <div
        className="ant-form-item-control "
      />
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={undefined}
  >
    <div
      className="ant-form-item-control-wrapper ant-col-xs-24 ant-col-xs-offset-0 ant-col-sm-14 ant-col-sm-offset-4"
    >
      <div
        className="ant-form-item-control "
      />
    </div>
  </div>
</div>
`;
